multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -24,20 +31,672 @@ use node_lib::database::db_initializer::{
+                );
+            }
+
+            (serving_node_1, serving_node_2, serving_node_3)
+        };
+
+    test_body(
+        test_global_config,
+        stimulate_payments,
+        start_serving_nodes_and_run_check,
+    );
+}
+
+#[test]
+fn payments_were_adjusted_due_to_insufficient_balances() {
+    let consuming_node_ui_port = find_free_port();
+    let serving_node_1_ui_port = find_free_port();
+    let serving_node_2_ui_port = find_free_port();
+    let serving_node_3_ui_port = find_free_port();
+    let payment_thresholds = PaymentThresholds {
+        threshold_interval_sec: 2_500_000,
+        debt_threshold_gwei: 100_000_000,
+        payment_grace_period_sec: 85_000,
+        maturity_threshold_sec: 85_000,
+        permanent_debt_allowed_gwei: 10_000_000,
+        unban_below_gwei: 1_000_000,
+    };
+    let owed_to_serv_node_1_minor =
+        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei + 5_000_000);
+    let owed_to_serv_node_2_minor =
+        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei + 20_000_000);
+    let owed_to_serv_node_3_minor =
+        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei + 60_000_000);
+    // Assuming all Nodes rely on the same set of payment thresholds
+    let cons_node_initial_service_fee_balance_minor = (owed_to_serv_node_1_minor
+        + owed_to_serv_node_2_minor)
+        - gwei_to_wei::<u128, u64>(2_345_678);
+    let agreed_transaction_fee_unit_price_major = 60;
+    let transaction_fee_needed_to_pay_for_one_payment_major = {
+        // We will need less but this should be accurate enough
+        let txn_data_with_maximized_cost = [0xff; 68];
+        let gas_limit_dev_chain = {
+            let const_part = web3_gas_limit_const_part(Chain::Dev);
+            u64::try_from(compute_gas_limit(
+                const_part,
+                txn_data_with_maximized_cost.as_slice(),
+            ))
+            .unwrap()
+        };
+        let transaction_fee_margin = Percentage::new(15);
+        transaction_fee_margin
+            .add_percent_to(gas_limit_dev_chain * agreed_transaction_fee_unit_price_major)
+    };
+    eprintln!(
+        "Computed transaction fee: {}",
+        transaction_fee_needed_to_pay_for_one_payment_major
+    );
+    let cons_node_transaction_fee_balance_minor =
+        2 * gwei_to_wei::<u128, u64>(transaction_fee_needed_to_pay_for_one_payment_major);
+    let test_global_config = TestInputsOutputsConfig {
+        ui_ports_opt: Some(Ports {
+            consuming_node: consuming_node_ui_port,
+            serving_node_1: serving_node_1_ui_port,
+            serving_node_2: serving_node_2_ui_port,
+            serving_node_3: serving_node_3_ui_port,
+        }),
+        // Should be enough only for two payments, the least significant one will fall out
+        cons_node_initial_transaction_fee_balance_minor_opt: Some(
+            cons_node_transaction_fee_balance_minor + 1,
```

Comment:
I also think it's safe to use `cons_node_transaction_fee_balance_minor` without adding any value.